pkg/virt-controller/watch/vmi.go

```diff
@@ -1339,17 +1358,10 @@ func isTempPod(pod *k8sv1.Pod) bool {
 	return ok
 }
 
-func getHotplugVolumes(vmi *virtv1.VirtualMachineInstance, virtlauncherPod *k8sv1.Pod) []*virtv1.Volume {
+func getHotplugVolumes(vmi *virtv1.VirtualMachineInstance) []*virtv1.Volume {
 	hotplugVolumes := make([]*virtv1.Volume, 0)
-	podVolumes := virtlauncherPod.Spec.Volumes
-	vmiVolumes := vmi.Spec.Volumes
-
-	podVolumeMap := make(map[string]k8sv1.Volume)
-	for _, podVolume := range podVolumes {
-		podVolumeMap[podVolume.Name] = podVolume
-	}
-	for _, vmiVolume := range vmiVolumes {
-		if _, ok := podVolumeMap[vmiVolume.Name]; !ok && (vmiVolume.DataVolume != nil || vmiVolume.PersistentVolumeClaim != nil) {
+	for _, vmiVolume := range vmi.Spec.Volumes {
+		if (vmiVolume.DataVolume != nil && vmiVolume.DataVolume.Hotpluggable) || (vmiVolume.PersistentVolumeClaim != nil && vmiVolume.PersistentVolumeClaim.Hotpluggable) {
 			hotplugVolumes = append(hotplugVolumes, vmiVolume.DeepCopy())
 		}
 	}
```

Comment:
There's something we need to think through here with regards to how we handle updates.

Before your PR here, the previous behavior was to cross reference the vmi struct and the vmi pod to discover the delta between the two. 

With your PR, hotplugged volumes are marked on the VMI spec.

Now, think about what happens to existing VMIs when the kubevirt control plane is updated from a previous version to the version that has your changes. VMIs  that started before the update that have hotplug volumes will not have the `HotPluggable` boolean set because that only happens in our virt-api subresource in the new version of KubeVirt.

the result is that old VMIs before the update that have hotplugable disks won't behave as we'd expect unless we preserve this logic of comparing the VMI pod with the VMI spec and calculating the delta.

Reply:
I found that there is no need to derive hotplug volumes from `vmi.spec`. The original solution is sufficient. So I removed it as well as the similar logic in vmi-update-admitter and host-disk. The only place need to concern `Hotpluggable` flag in `vmi.spec` is VMI template rendering (20215d2),  which should not include hotplug volumes when starting a new VMI.
After these changes, it should be backward compatible. 🙂

Reply:
Hi. I've addressed this. Feel free to review. Thanks!

Reply:
excellent, thanks. i'll give this another run through soon